src/status_im/ui/components/bottom_sheet/view.cljs

```diff
@@ -90,75 +83,116 @@
+                                   :style          (styles/content-container window-height sheet-height bottom-value)}
+                                  (when-not (or disable-drag?
+                                                (= max-height sheet-height))
+                                    (pan-handlers
+                                     (swipe-pan-responder {:bottom-value  bottom-value
+                                                           :opacity-value opacity-value
+                                                           :height        height
+                                                           :close-sheet   #(close-sheet)}))))
+             [react/view (merge {:style styles/content-header}
+                                (when-not disable-drag?
+                                  (pan-handlers
+                                   (swipe-pan-responder {:bottom-value  bottom-value
+                                                         :opacity-value opacity-value
+                                                         :height        height
+                                                         :close-sheet   #(close-sheet)}))))
+              (when show-handle?
+                [react/view styles/handle])]
+             [react/animated-view {:style {:height sheet-height}}
+              ;; NOTE: For a better UX on onScrollBeginDrag we can start dragging the sheet.
+              [react/scroll-view {:bounces        false
+                                  :scroll-enabled true
+                                  :style          {:flex 1}}
+               [react/view {:style     {:padding-top    styles/vertical-padding
+                                        :padding-bottom (+ styles/vertical-padding
+                                                           (:bottom safe-area))}
+                            :on-layout #(->> %
+                                             .-nativeEvent
+                                             .-layout
+                                             .-height
+                                             (reset! content-height))}
+                [content]]]]]]])))))
 
-                (and old-show? (false? new-show?) (true? @show-sheet?))
-                (cancel opts'))))
-      :reagent-render
-      (fn [{:keys [content content-height]}]
-        (let [total-content-height (+ content-height
-                                      styles/border-radius
-                                      styles/bottom-padding)]
-          (when @show-sheet?
-            [bottom-sheet-view (assoc opts
-                                      :content content
-                                      :height total-content-height)])))})))
+(defn bottom-sheet [props]
+  [react/safe-area-consumer
+   (fn [insets]
+     (reagent/as-element
+      [bottom-sheet-view (assoc props
+                                :window-height @(re-frame/subscribe [:dimensions/window-height])
```

Comment:
better to use `letsubs`

Reply:
With the upgrade of reagent and re-frame I think this macro doesn't make anymore sens for components. If we would like to make it just shorter we can use the largely adopted by the community composition of deref and subscribe `<sub` [lamdaisland](https://lambdaisland.com/blog/2017-02-11-re-frame-form-1-subscriptions) which is just pure function. While keeping the defview for screens where we can add other screen-related meta information like `on-focus` etc.

Reply:
but i still think its better to move subscriptions outside renderer even if we don't want to use letsubs, `@(re-frame/subscribe` in renderer function doesn't look right

Reply:
@flexsurfer it is fine to have subs in renderer function, we don't really need `letsubs` for quite a while already https://github.com/day8/re-frame/issues/218#issuecomment-264691621

Reply:
i know , but personaly i prefer letsubs, more clean and idiomatic , but its up to you guys how you want to use it